source/common/stats/lazy_init.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include "source/common/common/thread.h"
+
+namespace Envoy {
+namespace Stats {
+
+template <typename StatsStructType> class LazyableInterface {
```

Comment:
s/Lazyable/LazyCompatible/

doc class saying why this is needed (reference lazy-init bootstrap config)